inline-caching/README.md

```diff
@@ -0,0 +1,255 @@
+Our CacheLoader implementation also uses JOOQ for connecting to Postgres and Java system properties to retrieve Postgres' username and password.
+
+### AsyncEventListener
+Gemfire's AsyncEventListener is used to listen for events to the cache, and then perform a batch update to the backing postgres database.
+HikariCP, a popular connection pooling library for Java applications, is being used here to help handle our postgres connections. We also
+are using a singleton pattern in order to load only one connection pool in our cache. 
+
+### Create a region with an AsyncEventListener and a CacheLoader
+
+To create a region with an AsyncEventListener and a CacheLoader using GemFire, you can use `gfsh`. 
+Be sure that you have already built our example project and downloaded the necessary JDBC drivers as outlined above. 
+We will be creating a GemFire region with the name `item`.
+```bash
+./<location of gemfire>/bin/gfsh
+start locator --name locator
+start server --name='server' --classpath="<directory for postgres jar downloaded above>/postgresql-42.6.0.jar:<project root directory>/cache-loader/build/libs/*" --J="-Dpostgres.username=myuser" --J="-Dpostgres.password=mypass"
+create async-event-queue --listener=io.vmware.event.ItemAsyncEventListener --id=item-writebehind-queue --batch-size=10 --batch-time-interval="20"
+create region --name=item --type=REPLICATE --cache-loader=io.vmware.event.ItemCacheLoader --async-event-queue-id=item-writebehind-queue
+```
+
+Notice that the jars are passed into the classpath for our JDBC Postgres driver and the jar that contains our CacheLoader and AsyncEventListener implementations.
+These classes will be used by GemFire whenever we interact with our newly created `item` region. Our new region also utilizes the options `--cache-loader` and `--async-event-queue-id`. 
+These point to the implementation of the AsyncEventListener and CacheLoader that were referenced earlier.
+
+## Define the data model using Spring Data
+
+The Spring Data model that will be used to represent data in GemFire typically involves defining a set of Java classes that represent the Spring Data model entities.
+For our example, we are just going to use a simple String, but any Java object can be used. 
+
+Here is an example for how to define a Spring Data model using GemFire's repositories that is mapped to our `item` region that we created above
+
+```java
+@Region("/item")
+public interface Repository extends GemfireRepository<String, String> {
+}
+```
+
+Again, we are using Java Strings here for simplicity, but we can use any Java object to store data in GemFire. 
+
+## Web Project
+
+Our `web` project consists of 4 objects:
+1. Application - spring configuration for our app
+2. Controller - handles the web traffic and interacts with the Service
+3. Repository - defines key and value of our data model
+4. Service - uses a Repository and can utilize business logic to adapt GemFire data into a nicer format 
```

Comment:
These need to be adjusted to match the changes in class names

Reply:
adjusted